3ds2/src/main/java/com/adyen/checkout/adyen3ds2/ui/DAAuthenticationView.kt

```diff
@@ -0,0 +1,221 @@
+        binding.btnAuthoriseDifferently.text = localizedContext.getString(
+            R.string.checkout_3ds2_da_authentication_negative_button
+        )
+        setRemoveCredentialsLocalizedText(localizedContext)
+    }
+
+    private fun setRemoveCredentialsLocalizedText(localizedContext: Context) {
+        val removeCredentialsText = localizedContext.getText(
+            R.string.checkout_3ds2_da_authentication_remove_credentials
+        ) as SpannedString
+
+        val removeCredentialsStringBuilder = SpannableStringBuilder(removeCredentialsText)
+        val annotations = removeCredentialsText.getSpans(
+            0,
+            removeCredentialsText.length,
+            android.text.Annotation::class.java
+        )
+        annotations?.indices?.forEach { i ->
+            val annotation = annotations[i]
+            val annotationKey = annotation.key
+            if (annotationKey.equals("link")) {
+                val linkTextColor = ContextCompat.getColor(localizedContext, R.color.textColorLink)
+                removeCredentialsStringBuilder.setSpan(
+                    ForegroundColorSpan(linkTextColor),
+                    removeCredentialsText.getSpanStart(annotation),
+                    removeCredentialsText.getSpanEnd(annotation),
+                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+                )
+            }
+        }
+        binding.tvRemoveCredentials.text = removeCredentialsStringBuilder
+    }
+
+    private fun collectTimerUpdates(activity: Activity, coroutineScope: CoroutineScope, delegate: Adyen3DS2Delegate) {
+        coroutineScope.launch {
+            delegate.getAuthenticationTimerFlow().collect {
+                withContext(Dispatchers.Main) {
+                    setTimerData(it)
+                }
+                if (it.millisUntilFinished == 0L) {
+                    delegate.onAuthenticationResult(DAAuthenticationResult.Timeout, activity)
+                }
+            }
+        }
+    }
+
+    private fun setTimerData(timerData: TimerData) {
+        val minutes = timerData.millisUntilFinished.milliseconds.inWholeMinutes
+        val seconds = timerData.millisUntilFinished.milliseconds.inWholeSeconds % 1.minutes.inWholeSeconds
+
+        val timeLeftString = localizedContext.getString(
+            R.string.checkout_3ds2_time_left_to_approve,
+            minutes,
+            seconds
+        )
+
+        binding.tvTimeLeft.text = timeLeftString
+        binding.lpiProgress.progress = timerData.progress
+    }
+
+    private fun showRemoveCredentialsDialog(context: Context, onPositiveClick: () -> Unit) {
+        val onClickListener = DialogInterface.OnClickListener { dialog, which ->
+            when (which) {
+                DialogInterface.BUTTON_POSITIVE -> onPositiveClick.invoke()
+            }
+            dialog.dismiss()
+        }
+        MaterialAlertDialogBuilder(context, R.style.DAMaterialAlertDialogTheme)
```

Comment:
It was used to follow the figma design, however as we already use one type of dialog in the project - it will be better to save consistency, I'll replace it to `AlertDialog.Builder`